### ISSUE TITLE
Implementing auto save is confusing

### ISSUE THREAD

**Issue by kgish** [abuiles/ember-cli-101-errata#59]
I found the whole discussion about Implementing auto save very confusing.

There's too much packed into a limited number of lines and more effort should be spent in clarifying concepts better. For example:
- Why use `this.send('save', this.get('model'))` when you could just as easily use `model.save()`?
- Why skip auto save when `isNew` is true? According to the documentation `isNew` is true once a record has been created until the first time it is saved.
- What does `By default, observers are not set up until the function where they are specified is consumed` mean? What is `consumed`?
- In the `observer example` the purpose of `Ember.run.once` is to queue and collate multiple requests into a single action. So for example you could even change parameter a and b many times whereas only the last stable state is required for firing off the observer.

(Sorry if I sound too critical, but that's not my intention. The truth is that you have a really good book going here and I'd just like to help you make it even better).

**Comment by abuiles** [comment]
@kgish thank you very much! I really appreciate your comments :) so no worries, they have helped a lot, I'll think about the things you are pointing here and update the book accordingly.

**Comment by alanbhamilton** [comment]
I was also finding it hard to follow the auto save section. What was confusing to me was that I was expecting to see `autoSave` run every time I changed the value in the select, and this was not happening. I think this has something to do with how `Ember.run.once` works but I'm not sure. I also tried changing it to `Ember.run` and saw that `isDirtyChanged` was fired twice each time the select value changed--resulting in two separate identical saves to the server.

To fix all this I just ripped out `autoSave` and `isDirtyChanged` then modified app/templates/articles/index.hbs as follows:

```
<td {{action 'save' model on='change'}}>{{view select content=states selection=model.state}}</td>
```

This works as we already have a `save` function in the articles index route.
